wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -73,6 +67,24 @@ newPayment sa passphrase srcAccount dstAccount coin policy = do
         (one (dstAccount, coin))
         policy
 
+newPaymentBatch
+    :: MonadWalletWebMode m
+    => SendActions m
+    -> PassPhrase
+    -> NewPaymentBatchInit
+    -> m CTx
+newPaymentBatch sa passphrase NewPaymentBatchInit {..} = do
+    src <- decodeCTypeOrFail npbFrom
+    mDests <- mapM (sequence . second decodeCTypeOrFail) npbTo
+    dst <- nonEmpty mDests `whenNothing`
+        throwM (RequestError "At least one recipient required.")
+    sendMoney
+        sa
+        passphrase
+        (AccountMoneySource src)
+        dst
```

Comment:
@Martoon-00 In the relevant issue, you said:

> "Note that things are not so easy here. Till this moment we didn't yet thought thoroughly about many-outputs case. At least we have to entirely revise how do we form transaction history and write bunches of tests for it."

Mind elaborating a little bit further? Do you think we are exposing ourselves to problems we aren't seeing?

Reply:
I don't think that we should _entirely_ revise how do we form transaction history. Code in `Pos.Wallet.Web.Tracking` module which forms transaction history cache works without assumptions about number of outputs or belonging of all outputs to one wallet. Transaction is classified as _incoming_ or _outgoing_ simply by counting how transaction changes the balance of the wallet (sum of all outputs belonging to own addresses minus sum of all inputs belonging to own addresses); transaction is _incoming_ if balance difference is positive and _outgoing_ otherwise.

I just wanted to say that we seemingly wrote all relevant logic with possibility of multi-output transactions in mind, so I don't believe it will turn out to be a huge problem. Though I agree that we should provide tests for it.